examples/examples_test.go

```diff
@@ -586,6 +586,594 @@ func TestWrongStateMaxItemOneDiffProduced(t *testing.T) {
+            ],
+            "environment": {
+                "__defaults": [],
+                "variables": {
+                    "__defaults": [],
+                    "foo": "bar"
+                }
+            },
+            "ephemeralStorage": {
+                "__defaults": [],
+                "size": 512
+            },
+            "handler": "index.test",
+            "loggingConfig": {
+                "__defaults": [],
+                "logFormat": "Text",
+                "logGroup": "/aws/lambda/testLambda-83906f2"
+            },
+            "name": "testLambda-83906f2",
+            "packageType": "Zip",
+            "role": "arn:aws:iam::616138583583:role/iamForLambda-d5757fe",
+            "runtime": "nodejs18.x",
+            "sourceCodeHash": "WUsPYQdwiMj+sDZzl3tNaSzS42vqVfng2CZtgcy+TRs=",
+            "tracingConfig": {
+                "__defaults": [],
+                "mode": "PassThrough"
+            }
+        },
+        "randomSeed": "MpcMRlpQV7R8mD8Nmx9KpLPPIyFTHsB8HA4kxXdWTfo="
+    },
+    "response": {
+        "inputs": {
+            "__defaults": [
+                "memorySize",
+                "publish",
+                "reservedConcurrentExecutions",
+                "skipDestroy",
+                "timeout"
+            ],
+            "architectures": [
+                "x86_64"
+            ],
+            "environment": {
+                "__defaults": [],
+                "variables": {
+                    "__defaults": [],
+                    "foo": "bar"
+                }
+            },
+            "ephemeralStorage": {
+                "__defaults": [],
+                "size": 512
+            },
+            "handler": "index.test",
+            "loggingConfig": {
+                "__defaults": [
+                    "applicationLogLevel",
+                    "systemLogLevel"
+                ],
+                "applicationLogLevel": "",
+                "logFormat": "Text",
+                "logGroup": "/aws/lambda/testLambda-83906f2",
+                "systemLogLevel": ""
+            },
+            "memorySize": 128,
+            "name": "testLambda-83906f2",
+            "packageType": "Zip",
+            "publish": false,
+            "reservedConcurrentExecutions": -1,
+            "role": "arn:aws:iam::616138583583:role/iamForLambda-d5757fe",
+            "runtime": "nodejs18.x",
+            "skipDestroy": false,
+            "sourceCodeHash": "WUsPYQdwiMj+sDZzl3tNaSzS42vqVfng2CZtgcy+TRs=",
+            "timeout": 3,
+            "tracingConfig": {
+                "__defaults": [],
+                "mode": "PassThrough"
+            }
+        }
+    },
+    "metadata": {
+        "kind": "resource",
+        "mode": "client",
+        "name": "aws"
+    }
+}]`)
+}
+
+func TestSourceCodeHashUpdatedLambdaFails(t *testing.T) {
+  // Update requires a Configure, otherwise the AWS provider segfaults.
+
+  // Error looks like this but contains a request ID so we can't match it.
```

Comment:
Note that this correctly fails but the error is not at all readable. I'm wondering if there is a nice way for this to fail earlier - essentially we want to fail if a user inputs this and then updates the value.